src/Builders/Paths/RouteParams/RouteParamsBuilder.php

```diff
@@ -7,22 +7,35 @@
 use GoldSpecDigital\ObjectOrientedOAS\Objects\Parameter;
 use GoldSpecDigital\ObjectOrientedOAS\Objects\Schema;
 use Illuminate\Routing\Route;
+use LaravelJsonApi\Core\Server\Server;
+use LaravelJsonApi\Laravel\Routing\Route as JsonApiRoute;
 
 class RouteParamsBuilder
 {
     /** @return Parameter[] */
-    public static function build(Route $route): array
+    public static function build(Server $server, Route $route): array
     {
         preg_match_all('/{[A-z0-9]+}/', $route->uri(), $routeParamNames, PREG_PATTERN_ORDER);
         $routeParams = [];
         foreach ($routeParamNames[0] as $routeParamName) {
             $routeParams[] = (new Parameter())
                 ->in('path')
                 ->name(trim($routeParamName, '{}'))
-                ->schema((new Schema())->type('integer'))
+                ->schema(Schema::integer())
                 ->required();
         }
 
+	$resourceType = $route->parameter(JsonApiRoute::RESOURCE_TYPE);
+        if ($resourceType !== null && str_contains($route->getName(), 'index')) {
+            $schema = $server->schemas()->schemaFor($resourceType);
+            foreach ($schema->filters() as $filter) {
+                $routeParams[] = (new Parameter())
+                    ->in('query')
+                    ->name('filter[' . $filter->key() . ']')
+                    ->schema($filter->isSingular() ? Schema::string() : Schema::array());
```

Comment:
Laatste opmerking: Als ik het resultaat in de swagger editor gooi krijg ik fouten over missende item properties. Dit is op te lossen door de ->items methode aan te roepen op de Schema::array(). 
```suggestion
                    ->schema($filter->isSingular() ? $stringOrInteger : Schema::array()->items($stringOrInteger));
```

Met dit boven de foreach:
```diff
+$stringOrInteger = (new AnyOf())->schemas(Schema::string(), Schema::integer());
```